kclvm/sema/src/core/symbol.rs

```diff
@@ -0,0 +1,389 @@
+use generational_arena::Arena;
+use indexmap::IndexMap;
+
+use kclvm_error::Position;
+
+use super::package::{PackageDB, PackageInfo};
+
+#[derive(Default, Debug)]
+pub struct KCLSymbolData {
```

Comment:
Would U pls can add more documents or link the design doc into the issue https://github.com/kcl-lang/kcl/issues/742